### PR TITLE
Remove @​deprecated stuff

### LINKED ISSUE #3508: Remove all deprecated functionality from 3.0
This issue can be used as a reminder and an index of concrete issues:

1. #3518 — Remove `@​deprecated` stuff.
2. #3579 — Remove more `@​deprecated` stuff.
3. #3804 — Remove some leftovers.

### PR DESCRIPTION
**Pull request by Majkl578**
|      Q       |   A
|------------- | -----------
| Type         | improvement
| BC Break     | yes

#### Summary

Removes so far @​deprecated stuff for DBAL 3.0. Fixes #3508.

### CODE REVIEW

**Comment by Majkl578** [comment]
@morozov There will be conflicts due to changes in Type so we can wait until develop is up-to-date and I'll rebase the PR afterwards.

File: lib/Doctrine/DBAL/SQLParserUtils.php [hunk 1]
```diff
@@ -26,40 +26,15 @@
...
-    public const NAMED_TOKEN      = '(?<!:):[a-zA-Z_][a-zA-Z0-9_]*';
-    /**#@-*/
+    private const POSITIONAL_TOKEN = '\?';
+    private const NAMED_TOKEN      = '(?<!:):[a-zA-Z_][a-zA-Z0-9_]*';
 
     // Quote characters within string literals can be preceded by a backslash.
     public const ESCAPED_SINGLE_QUOTED_TEXT   = "(?:'(?:\\\\\\\\)+'|'(?:[^'\\\\]|\\\\'?|'')*')";
     public const ESCAPED_DOUBLE_QUOTED_TEXT   = '(?:"(?:\\\\\\\\)+"|"(?:[^"\\\\]|\\\\"?)*")';
     public const ESCAPED_BACKTICK_QUOTED_TEXT = '(?:`(?:\\\\\\\\)+`|`(?:[^`\\\\]|\\\\`?)*`)';
     private const ESCAPED_BRACKET_QUOTED_TEXT = '(?<!\b(?i:ARRAY))\[(?:[^\]])*\]';
```
Comment (morozov) [inline lib/Doctrine/DBAL/SQLParserUtils.php]: These constants should also be deprecated and made private.

**Review by morozov — COMMENTED** [review]
Looks good so far. A few notes:

1. `Configuration::setFilterSchemaAssetsExpression()` is also deprecated and should be removed.
2. There are a few cases with `trigger_error(/* ... */, E_DEPRECATED)`.

**Comment by morozov** [comment]
Also, `SQLSrvStatement::LAST_INSERT_ID_SQL` could be deprecated and made private.

**Comment by Majkl578** [comment]
Rebased.

File: tests/Doctrine/Tests/DBAL/SQLParserUtilsTest.php [hunk 2]
```diff
@@ -92,15 +92,6 @@ public function dataGetPlaceholderPositions()
         ];
     }
 
-    /**
-     * @dataProvider dataGetPlaceholderPositions
-     */
-    public function testGetPlaceholderPositions($query, $isPositional, $expectedParamPos)
```
Comment (morozov) [inline tests/Doctrine/Tests/DBAL/SQLParserUtilsTest.php]: Should this be reworked in two tests, not deleted?

Comment (Majkl578) [reply tests/Doctrine/Tests/DBAL/SQLParserUtilsTest.php]: Seems to be testing a deleted method.

Comment (morozov) [reply tests/Doctrine/Tests/DBAL/SQLParserUtilsTest.php]: Yes, but the method was split apart into two other methods, not just deleted.

Comment (Majkl578) [reply tests/Doctrine/Tests/DBAL/SQLParserUtilsTest.php]: Hmm but the test wasn't? That's weird. Ok.

Comment (Majkl578) [reply tests/Doctrine/Tests/DBAL/SQLParserUtilsTest.php]: Ah, the methods are private.
How do you suggest to test them? Reflection?

Comment (Majkl578) [reply tests/Doctrine/Tests/DBAL/SQLParserUtilsTest.php]: Split into two new tests and two data providers, using reflection - if you have better idea let me know. :)

Comment (morozov) [reply tests/Doctrine/Tests/DBAL/SQLParserUtilsTest.php]: > Hmm but the test wasn't? That's weird. Ok.

I didn't notice it at that time.

> […] if you have better idea let me know. :)

It's good enough to me.

**Comment by Majkl578** [comment]
Why was this PR closed? @morozov @jwage?

**Comment by jwage** [comment]
Work is being continued here https://github.com/doctrine/dbal/pull/3538

**Comment by Majkl578** [comment]
That is not a valid reason to close an open PR that was not inactive - no one even asked me to rebase this PR or anything else. You are basically only stealing my work.

**Comment by jwage** [comment]
@Majkl578 Happy to continue the work in here. Since you closed many other PRs and deleted the branches I assumed you were not interested in continuing the work. Apologies if that assumption was incorrect. Would you like to continue the work here?

**Comment by Majkl578** [comment]
I closed PRs in other repos because I lost interest in contributing to an organization that treats members like gabrage.
Regarding PRs in DBAL @morozov and me came up with a list of issues to either finish or abandon, those open are active and to be finished either by me, by @morozov or both.

**Comment by jwage** [comment]
Understood. @morozov asked if I could finish #3518 and I obliged. Would you like to finish the pull request here or make the branch open so that we can continue the work?

**Comment by Majkl578** [comment]
I will do it here but at this moment there are no requested changes here - the PR was simply ready for review/merge (except maybe rebase), not aware of anything needed to _finish #3518_.

**Comment by jwage** [comment]
Cool. Can you rebase? Then @morozov and I will help get it reviewed and merged.

**Comment by Majkl578** [comment]
Sure, will do soon.

**Comment by morozov** [comment]
Thanks @Majkl578!

**Pull request merged**